### PR TITLE
Blr/code formatting cleanup

### PR DESCRIPTION
**Pull request by bartriordan**
@BJK @maneeshnycda Please review. All I focused on was cleaning up spelling, grammar, and code formatting. All tests are passing. This PR does not include a new build since it does not change the functionality of the library.

### CODE REVIEW

File: src/lib/jsonFetcher.js [hunk 1]
```diff
@@ -14,73 +19,66 @@ var fetcher = {
...
+      opts.xhrFields = {withCredentials: true}
+    return opts
   },
   sendRequest: function(request) {
-    var promise = jQuery.Deferred();
+    var promise = jQuery.Deferred()
     jQuery.ajax(request)
-      .done(function (data, result, response){
-        if (typeof(data) != "object")
-          data = JSON.parse(data); // this would only be a string if a test fixture returns it -- would love to not have test side effects here -BJK
```
Comment (BJK) [inline src/lib/jsonFetcher.js]: would like to keep this comment

File: src/lib/jsonFetcher.js [hunk 2]
```diff
@@ -14,73 +19,66 @@ var fetcher = {
...
-    var request = this.buildRequest(url, 'POST', data);
+    // Taken from: http://stackoverflow.com/questions/12431760/html5-formdata-file-upload-with-rubyonrails
+    // and http://stackoverflow.com/questions/21234106/upload-file-using-reactjs-via-blueimp-fileupload-jquery-plugin
+    var request = this.buildRequest(url, 'POST', data)
     request.cache = false
     request.processData = false
     request.contentType = false
-    return this.sendRequest(request);
-  },
-  parseError: function(err) {
```
Comment (BJK) [inline src/lib/jsonFetcher.js]: why removed?

File: src/lib/jsonFetcher.js [hunk 3]
```diff
@@ -1,12 +1,17 @@
...
-var _cleanSlashes = function(path) {
-  if (path.indexOf("/") === 0)
-    return path;
-  else
-    return "/" + path;
+function cleanSlashes(path) {
+  return `${path.indexOf('/') === 0 ? '' : '/'}${path}`
+}
+
+// TODO: Success data will only be a string if a test fixture returns it, and test side effects should not dictate the
```
Comment (bartriordan) [inline src/lib/jsonFetcher.js]: @BJK I just moved your comment up here as a more-noticeable TODO. I can move it back to where it was if you prefer.

File: src/lib/jsonFetcher.js (hunk 2 above)
Comment (bartriordan) [inline src/lib/jsonFetcher.js]: @BJK It was never used or passed anywhere. I figured if we ever need it we can just find it again with `git log -S'parseError'`.

**Comment by bartriordan** [comment]
@BJK Fixed conflicts with master and responded to your comments.

File: src/lib/jsonFetcher.js (hunk 3 above)
Comment (BJK) [inline src/lib/jsonFetcher.js]: yeah, in context is best.

File: src/lib/StorePrototype.js [hunk 4]
```diff
@@ -1,116 +1,105 @@
...
+    addChangeListener: function(callback) {
+      this.on('CHANGE', callback)
+    },
+    removeChangeListener: function (callback) {
+      this.removeListener('CHANGE', callback)
+    },
+    getMessage: function() {
+      return this.message
+    },
+    setMessage: function(message) {
```
Comment (BJK) [inline src/lib/StorePrototype.js]: this will have some consequences when we make the EP codebase use this.

**Comment by BJK** [comment]
I'm good to merge this. The only thing to be aware of, is that some of the files have since moved further along in the EP codebase. We'll need to do an assessment, and you make need to 'redo' some of this clean up. But nothing really to do about that.

**Pull request merged**